### PR TITLE
add explaination about `nested-loop-join-cache-capacity`

### PR DESCRIPTION
**Pull request by Reminiscent**
### What is changed, added or deleted? (Required)
add explaination about a new system variable `nested-loop-join-cache-capacity` to tidb-configuration-file.md

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- https://github.com/pingcap/tidb/pull/17039

### CODE REVIEW

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -349,6 +349,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：false
 + 该变量作为系统变量 [`tidb_opt_distinct_agg_push_down`](/tidb-specific-system-variables.md#tidb_opt_distinct_agg_push_down) 的初始值。
 
+### `nested-loop-join-cache-capacity`
+
++ nested loop join cache LRU 使用的最大内存限制。可以占用的最大内存阈值，单位为字节。
++ 默认值：20971520
++ 当 `nested-loop-join-cache-capacity = 0` 时，默认关闭 nested loop join cache. 当 LRU 的 size 大于 `nested-loop-join-cache-capacity` 时，也会剔除 LRU 中的元素。
```
Comment (TomShawn) [inline tidb-configuration-file.md:356]:
```suggestion
+ 当 `nested-loop-join-cache-capacity = 0` 时，默认关闭 nested loop join cache。 当 LRU 的 size 大于 `nested-loop-join-cache-capacity` 时，也会剔除 LRU 中的元素。
```

**Comment by TomShawn** [comment]
@Reminiscent Does this change apply to other version(s)? If yes, please tick the boxes in the description part.

**Comment by Reminiscent** [comment]
> @Reminiscent Does this change apply to other version(s)? If yes, please tick the boxes in the description part.

No need to apply to other versions. Just in the master version.

**Comment by Reminiscent** [comment]
@TomShawn @qw4990 PTAL

**Pull request merged**